### PR TITLE
Add turn signal functionality and fixes

### PR DESCRIPTION
**Pull request by lbucklandAS**
Prior to this pull request there was a bug in the turn signal functionality for GEM. Also, there was no functionality in the teleop demo for turn signals. This fixes that bug and adds functionality.

### CODE REVIEW

**Comment by Capri2014** [comment]
@lbucklandAS Lint issue in //modules/canbus/tools:teleop_cpplint    before this can be merged to master

File: modules/canbus/tools/teleop.cc [hunk 1]
```diff
@@ -228,6 +234,24 @@ class Teleop {
...
                 << ", Brake = " << control_command_.brake();
           break;
+        case KEYCODE_SETQ1:
+        case KEYCODE_SETQ2:
+          if (read(kfd_, &c, 1) < 0) {
+            exit(-1);
+          }
+          static int cnt = 0;
+          cnt ++;
+          if (cnt > 2)
```
Comment (startcode) [inline modules/canbus/tools/teleop.cc]: Prefer using curly braces '{' after if, even there is only one statement.

File: modules/canbus/tools/teleop.cc [hunk 2]
```diff
@@ -228,6 +234,24 @@ class Teleop {
...
+            cnt = 0;
+
+          if (cnt == 0)
```
Comment (startcode) [inline modules/canbus/tools/teleop.cc]: Prefer using curly braces '{' after if, even there is only one statement.

File: modules/canbus/tools/teleop.cc [hunk 3]
```diff
@@ -228,6 +234,24 @@ class Teleop {
...
+            control_command_.mutable_signal()->set_turn_signal(VehicleSignal::TURN_NONE);
+          else if (cnt == 1)
```
Comment (startcode) [inline modules/canbus/tools/teleop.cc]: Prefer using curly braces '{' after if, even there is only one statement.

File: modules/canbus/tools/teleop.cc [hunk 4]
```diff
@@ -228,6 +234,24 @@ class Teleop {
...
+            control_command_.mutable_signal()->set_turn_signal(VehicleSignal::TURN_LEFT);
+          else if (cnt == 2)
```
Comment (startcode) [inline modules/canbus/tools/teleop.cc]: Prefer using curly braces '{' after if, even there is only one statement.

File: modules/canbus/tools/teleop.cc [hunk 5]
```diff
@@ -228,6 +234,24 @@ class Teleop {
           AINFO << "Throttle = " << control_command_.throttle()
                 << ", Brake = " << control_command_.brake();
           break;
+        case KEYCODE_SETQ1:
+        case KEYCODE_SETQ2:
+          if (read(kfd_, &c, 1) < 0) {
+            exit(-1);
+          }
+          static int cnt = 0;
+          cnt ++;
```
Comment (startcode) [inline modules/canbus/tools/teleop.cc]: Prefer '++cnt;'

**Comment by lbucklandAS** [comment]
@lianglia-apollo Thanks, this is great.

**Pull request merged**